modules/gateway/peers.go

```diff
@@ -52,8 +53,23 @@ func (g *Gateway) addPeer(p *peer) {
 	go g.listenPeer(p)
 }
 
+// randomPeer returns a random peer from the gateway's peer list.
+func (g *Gateway) randomPeer() (modules.NetAddress, error) {
+	if len(g.peers) > 0 {
+		r := rand.Intn(len(g.peers))
```

Comment:
our crypto package should probably have its own `Rand` function(s). It's confusing for `crypto` to refer to different packages throughout the codebase.